src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -0,0 +1,255 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+
+namespace GeneralizedHarmonic {
+
+template <size_t Dim>
+void ComputeDuDt<Dim>::apply(
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_psi,
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> dt_pi,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim>*> dt_phi,
+    const tnsr::aa<DataVector, Dim>& psi, const tnsr::aa<DataVector, Dim>& pi,
+    const tnsr::iaa<DataVector, Dim>& phi,
+    const tnsr::iaa<DataVector, Dim>& d_psi,
+    const tnsr::iaa<DataVector, Dim>& d_pi,
+    const tnsr::ijaa<DataVector, Dim>& d_phi, const Scalar<DataVector>& gamma0,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& gamma2,
+    const tnsr::a<DataVector, Dim>& gauge_function,
+    const tnsr::ab<DataVector, Dim>& spacetime_deriv_gauge_function,
+    const Scalar<DataVector>& lapse, const tnsr::I<DataVector, Dim>& shift,
+    const tnsr::II<DataVector, Dim>& upper_spatial_metric,
+    const tnsr::AA<DataVector, Dim>& upper_psi,
+    const tnsr::a<DataVector, Dim>& trace_christoffel,
+    const tnsr::abb<DataVector, Dim>& christoffel_first_kind,
+    const tnsr::Abb<DataVector, Dim>& christoffel_second_kind,
+    const tnsr::A<DataVector, Dim>& normal_spacetime_vector,
+    const tnsr::a<DataVector, Dim>& normal_spacetime_one_form) {
+  const size_t n_pts = shift.begin()->size();
+
+  // The claim in SpEC is that these temporaries are defined for optimisation.
```

Comment:
Do you want to file an issue about things to be aware of when profiling the GH RHS so we have a list of things that were done in the name of performance without measurement? The comment is effectively a 'TODO' which we are trying to avoid.